contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -105,6 +105,61 @@ contract DisputeManagerFacet is StateChannelCommon {
         return disputes[channelId];
     }
 
+    /// @dev create Block related Dispute
+    /// @param channelId - the channel id
+    /// @param proofs - fraud proof type (only block related fraud proofs are supported)
+
+    function createBlockDispute(
+        bytes32 channelId,
+        uint forkCnt,
+        Proof[] memory proofs,
+        ConfirmedBlock[] memory virtualVotingBlocks,
+        bytes memory encodedLatestFinalizedState,
+        bytes memory encodedLatestCorrectState
+    ) public {
+        require(!isDisputeInProgress(channelId), ErrorDisputeInProgrees());
+        require(getForkCnt(channelId) == forkCnt, ErrorDisputeForkMismatch());
+        address[] memory participants = getParticipants(channelId, forkCnt);
+
+        // state checks
+         require(
+            isFinalizedAndLatest(
+                channelId,
+                forkCnt,
+                encodedLatestFinalizedState,
+                encodedLatestCorrectState,
+                virtualVotingBlocks,
+                participants
+            ),
+            ErrorLatestFinalizedBlock()
```

Comment:
In general we don't do any of these checks anymore since they're expensive and don't actually solve the problems since on-chain we don't have the full off-chain context), so we let anyone commit to anything at the risk of slashing - and rely on crypto-economic game theory that the majority will be honest and that few that commit to junk will get slashed anyway by explicitly invoking auditing or generating more fraud proofs - if it's unclear how this should be implemented from the docs either ping me or I don't mind implementing this next week when I'm back from the PBA

There are some checks simple checks that we do as safety guards defined in the doc to protect the disputer from race condition on-chain